lib/foreman/renderer.rb

```diff
@@ -105,6 +105,18 @@ def indent(count)
       prefix + text.gsub(/\n/, "\n#{prefix}")
     end
 
+    def dns_lookup(name_or_ip)
+      resolver = Resolv::DNS.new
+      Timeout.timeout(Setting[:dns_conflict_timeout]) do
+        ipquery = IPAddr.new(name_or_ip) rescue false
```

Comment:
I tried with a variety of options, all of them fell under that class (most likely a subclass of it) of exception